c++/src/kj/compat/http.c++

```diff
@@ -6187,7 +6200,8 @@ private:
       auto pipe = newTwoWayPipe();
 
       kj::Own<kj::AsyncIoStream> io =
-          kj::heap<DelayedEofIoStream>(kj::mv(pipe.ends[0]), task.attach(kj::addRef(*this)));
+          kj::heap<DelayedEofIoStream>(kj::mv(pipe.ends[0]),
+              task.attach(kj::addRef(*this), kj::mv(headersCopy)));
```

Comment:
You are moving away `headersCopy` here but then you return it again below. You'll end up returning a null pointer. I guess we don't have any tests that actually examine the returned headers?

Reply:
Related discussion: https://github.com/capnproto/capnproto/pull/1567#discussion_r1020415427.

Can you confirm that this definitely needs to be attached to something? if so what can I attach it to in this code?

Reply:
Makes sense, I'll fix this once the http connect pipeline PR is merged.

Reply:
Just for future reference: This was resolved as part of the pipelining PR.